### PR TITLE
Bug #155: Encapsulate var manager and make it an instance of GDBBackend

### PR DESCRIPTION
**Pull request by jonahgraham**

### CODE REVIEW

**Comment by jonahgraham** [comment]
Hi @jcrwilliams - if you are still working on and familiar with the adapter I would appreciate a look at this please.

File: src/GDBBackend.ts [hunk 1]
```diff
@@ -36,10 +37,15 @@ export declare interface GDBBackend {
 
 export class GDBBackend extends events.EventEmitter {
     protected parser = new MIParser(this);
+    protected varMgr = new VarManager(this);
     protected out?: Writable;
     protected token = 0;
     protected proc?: ChildProcess;
 
+    public getVarMgr(): VarManager {
```
Comment (thegecko) [inline src/GDBBackend.ts]: Could use a getter for this? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

Comment (jonahgraham) [reply src/GDBBackend.ts]: I could, but then it wouldn't look like a Java programmer wrote it ;0)

Comment (jonahgraham) [reply src/GDBBackend.ts]: fixed in 120459b

**Comment by jonahgraham** [comment]
> Hi @jcrwilliams - if you are still working on and familiar with the adapter I would appreciate a look at this please.

@jcrwilliams I have merged this, but if you have any comments I can still fix it up.

**Pull request merged**